content/en/docs/pipelines/tutorials/benchmark-examples.md

```diff
@@ -0,0 +1,184 @@
+
+After a proper pipeline is chosen, the benchmark scripts will run it multiple
+times simultaneously as mentioned before. Among all the operations that the Kubeflow
+Pipelines can perform, running a pipeline is arguably the most unpredictable and
+costly one. Other operations, such as creating a pipeline (version) or creating an
+experiment, usually induce a predictable and moderate cost. For example, creating a
+pipeline version will introduce a new row in the pipeline versions table and a new
+file in minio server. The new file's size depends on the pipeline version's
+manifest. If we exclude the rare case of an extremely large manifest and assume
+an average sized manifest for each created pipeline version, the total cost of
+creating a pipeline version grows linearly with the number of pipeline versions.
+However, on the other hand, the cost of running a pipeline or a pipeline version
+involves much more uncertainty and sometimes quite a high cost. A pipeline or a
+pipeline version can have arbitrary components and hence running a pipeline or a
+pipeline version can incur arbitrary time and space complexities. For example, a step
+in a pipeline can use a customized container image which performs a super
+expensive training task. In addition, the runs in a Kubeflow Pipelines instance
+also consume more DB space than pipelines, pipeline versions, experiments, etc.
+Therefore, in order to understand the performance and scalability pain points in
+a Kubeflow Pipelines instance, it is more effective to focus on the run operation
+in workloads.
+
+## Prerequisites for running benchmark scripts
+
+To run the provided benchmark scripts, you need the following:
+
+*  A Jupyter notebook environment with access to the Kubeflow Pipelines API on
+   your Kubeflow Pipelines cluster. For example, you must be able to call the
+   CREATE, GET, DELETE, and LIST methods of the pipeline, pipeline version,
+   run, job, and experiment services from your Jupyter notebook environment.
+*  A Kubeflow Pipelines cluster. If you do not have a Kubeflow Pipelines
+   cluster, learn more about your [options for installing Kubeflow
+   Pipelines](/docs/pipelines/installation/overview/).
+*  A pipeline manifest. For example, this guide uses the
+   [taxi_updated_pool.yaml](https://storage.googleapis.com/jingzhangjz-project-pipelines/benchmarks/taxi_updated_pool.yaml)
+   pipeline manifest file.
+
+One way of setting up everything and running a benchmark script is shown below
+as an example.
+
+## Running the benchmark scripts
+
+Use the following instructions to run the benchmark script on your Kubeflow
+Pipelines cluster.
+
+1. Download the the [run_service_api.ipynb](https://github.com/jingzhang36/pipelines/blob/different_tools/tools/benchmarks/run_service_api.ipynb)
```

Comment:
Shouldn't this link go to kubeflow/pipelines instead of your fork?

Reply:
This is a little bit complicated. We already have a 'run_service_api.ipynb' in github. This is a variation of it. And this variation is made to run a special pipeline that is made to use two node pools. Let me try if I can move the special pipeline to ml-pipeline project and then check in this 'run_service_api.ipynb' as well.

Otherwise, if this is too confusing, I'll just use the basic pipeline with the existing 'run_service_api.ipynb'

Reply:
OK, SGTM.

Reply:
I moved the file run_service_api.ipynb to our public bucket at ml-pipeline https://storage.googleapis.com/ml-pipeline/sample-benchmark/run_service_api.ipynb